lib/src/io/js.dart

```diff
@@ -181,6 +181,21 @@ bool dirExists(String path) {
   });
 }
 
+bool linkExists(String path) {
+  if (!isNodeJs) {
+    throw UnsupportedError("linkExists() is only supported on Node.js");
+  }
+  return _systemErrorToFileSystemException(() {
+    try {
+      return fs.statSync(path).isSymbolicLink();
```

Comment:
```suggestion
      return fs.lstatSync(path).isSymbolicLink();
```

`stat` checks resolved symlink thus it always return `false` for `isSymbolicLink`. I think you want `lstat` which checks the symlink itself.